Add GKE support for tier 1 networking in GKE

kawych opened the pull request:
Add GKE support for tier 1 networking in GKE

Related issue: https://github.com/hashicorp/terraform-provider-google/issues/11588. It covers only thread per core, but I think adding tier1 networking support would be helpful for similar use cases.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Note:** unit tests seem to be broken. I was able to run `make test` for generated code as follows:

```
$ make test
$ make fmt
$ make test
...
google/resource_bigtable_app_profile.go:468:9: undefined: bigtableadmin
google/resource_compute_subnetwork.go:39:22: undefined: cidr
...
$ rm google/resource_bigtable_app_profile.go google/resource_bigtable_app_profile_test.go google/resource_compute_subnetwork.go google/resource_compute_subnetwork_test.go
$ sed -i '/resourceBigtableAppProfile/d' google/provider.go
$ sed -i '/resourceComputeSubnetwork/d' google/provider.go
$ make test
```

Release notes

```release-note:enhancement
container: added support for `network_performance_config.total_egress_bandwidth_tier` to support GKE tier 1 networking
```

modular-magician commented:
Hello!  I am a robot who works on Magic Modules PRs.

I've detected that you're a community contributor. @rileykarson, a repository maintainer, has been assigned to assist you and help review your changes. 

<details>
  <summary>:question: First time contributing? Click here for more details</summary>

---

Your assigned reviewer will help review your code by: 
* Ensuring it's backwards compatible, covers common error cases, etc.
* Summarizing the change into a user-facing changelog note.
* Passes tests, either our "VCR" suite, a set of presubmit tests, or with manual test runs.

You can help make sure that review is quick by running local tests and ensuring they're passing in between each push you make to your PR's branch. Also, try to leave a comment with each push you make, as pushes generally don't generate emails.

If your reviewer doesn't get back to you within a week after your most recent change, please feel free to leave a comment on the issue asking them to take a look! In the absence of a dedicated review dashboard most maintainers manage their pending reviews through email, and those will sometimes get lost in their inbox.

---

</details>

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 48 insertions(+), 4 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `0` 
Passed tests `0` 
Skipped tests: `0` 
Failed tests: `0` 

Errors occurred during REPLAYING mode. Please fix them to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/d4d84c68-1357-439b-a00d-b312cb5dbe89/build-log/replaying_test.log)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 48 insertions(+), 4 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2251` 
Passed tests `2009` 
Skipped tests: `241` 
Failed tests: `1` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig </blockquote></details>

modular-magician commented:
Tests failed during RECORDING mode:
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/309de5b6-5bd5-4b7a-a47a-1ebaad974d6c/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/309de5b6-5bd5-4b7a-a47a-1ebaad974d6c/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/309de5b6-5bd5-4b7a-a47a-1ebaad974d6c/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/309de5b6-5bd5-4b7a-a47a-1ebaad974d6c/recording) for each test

rileykarson commented on mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb:
```diff
@@ -2160,6 +2160,9 @@ resource "google_container_node_pool" "with_auto_pod_cidr" {
 	create_pod_range    = true
     pod_range           = "auto-pod-range"
 	pod_ipv4_cidr_block = "10.2.0.0/20"
+	node_performance_config = {
```
I think this should be `network_performance_config`?

kawych replied:
Yes, fixed

rileykarson commented on mmv1/third_party/terraform/resources/resource_container_node_pool.go.erb:
```diff
@@ -354,6 +354,21 @@ var schemaNodePool = map[string]*schema.Schema{
+						Schema: map[string]*schema.Schema{
+							"total_egress_bandwidth_tier": {
+								Type:        schema.TypeString,
+								Required:    true,
```
Given this field is updatable, can you add a test updating from having it enabled to disabled?

rileykarson replied:
Ah, this is the first field under `network_config` that's updatable. GKE's complex update methods mean that updates are nonstandard for these resources- we'll need to update `nodePoolUpdate` to look for changes in `network_config` and send an update if they exist! That function's represented as a set of update call blocks, gated on specific field groups- any similar block using the `UpdateNodePoolRequest` type should serve as a reference.

rileykarson commented on mmv1/third_party/terraform/resources/resource_container_node_pool.go.erb:
```diff
@@ -354,6 +354,21 @@ var schemaNodePool = map[string]*schema.Schema{
+					Description: `Network bandwidth tier configuration.`,
+					Elem:        &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"total_egress_bandwidth_tier": {
```
Would it make sense to add support for `externalIpEgressBandwidthTier` at the same time? (non-blocking)

kawych replied:
I see it's much less documented than total_egress_bandwidth_tier, I also played with it a little and got failures from GKE API, so I would prefer to stay with just this one.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 57 insertions(+), 4 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `0` 
Passed tests `0` 
Skipped tests: `0` 
Failed tests: `0` 

Errors occurred during REPLAYING mode. Please fix them to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/f553ba38-6283-4440-9bd9-e7790f65153b/build-log/replaying_test.log)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 57 insertions(+), 4 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2254` 
Passed tests `2012` 
Skipped tests: `241` 
Failed tests: `1` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig </blockquote></details>

modular-magician commented:
Tests failed during RECORDING mode:
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/e1fb8d06-10bb-429b-9d2e-71fcc9d58305/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/e1fb8d06-10bb-429b-9d2e-71fcc9d58305/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/e1fb8d06-10bb-429b-9d2e-71fcc9d58305/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/e1fb8d06-10bb-429b-9d2e-71fcc9d58305/recording) for each test

modular-magician commented:
#### Tests analytics 
Total tests: `2254` 
Passed tests `2012` 
Skipped tests: `241` 
Failed tests: `1` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig </blockquote></details>

modular-magician commented:
Tests failed during RECORDING mode:
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/196cbc81-1969-4f9a-92ca-ede62b503633/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/196cbc81-1969-4f9a-92ca-ede62b503633/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/196cbc81-1969-4f9a-92ca-ede62b503633/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/196cbc81-1969-4f9a-92ca-ede62b503633/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 89 insertions(+), 7 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 89 insertions(+), 7 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2256` 
Passed tests `2013` 
Skipped tests: `241` 
Failed tests: `2` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig|TestAccComputeForwardingRule_update </blockquote></details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccComputeForwardingRule_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/4cf43fe1-f30a-485e-b3db-fc8da32a2ee7/recording/TestAccComputeForwardingRule_update.log)] 

Tests failed during RECORDING mode:
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/4cf43fe1-f30a-485e-b3db-fc8da32a2ee7/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/4cf43fe1-f30a-485e-b3db-fc8da32a2ee7/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/4cf43fe1-f30a-485e-b3db-fc8da32a2ee7/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/4cf43fe1-f30a-485e-b3db-fc8da32a2ee7/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 89 insertions(+), 7 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2260` 
Passed tests `2017` 
Skipped tests: `241` 
Failed tests: `2` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig|TestAccFirebaserulesRelease_BasicRelease </blockquote></details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/7cb9b2b4-cc84-4ef2-8b32-99e971d0dc8a/recording/TestAccFirebaserulesRelease_BasicRelease.log)] 

Tests failed during RECORDING mode:
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/7cb9b2b4-cc84-4ef2-8b32-99e971d0dc8a/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/7cb9b2b4-cc84-4ef2-8b32-99e971d0dc8a/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/7cb9b2b4-cc84-4ef2-8b32-99e971d0dc8a/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/7cb9b2b4-cc84-4ef2-8b32-99e971d0dc8a/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 147 insertions(+), 59 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `0` 
Passed tests `0` 
Skipped tests: `0` 
Failed tests: `0` 

Errors occurred during REPLAYING mode. Please fix them to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/1d78478d-317d-43e4-a48d-575ccd5b3759/build-log/replaying_test.log)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 229 insertions(+), 134 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `0` 
Passed tests `0` 
Skipped tests: `0` 
Failed tests: `0` 

Errors occurred during REPLAYING mode. Please fix them to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/cb378242-e50a-497c-97d9-252713038d89/build-log/replaying_test.log)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 89 insertions(+), 8 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2264` 
Passed tests `2022` 
Skipped tests: `241` 
Failed tests: `1` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig </blockquote></details>

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 7 insertions(+), 1 deletion(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 78 insertions(+), 8 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6826-old..auto-pr-6826) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2268` 
Passed tests `2023` 
Skipped tests: `243` 
Failed tests: `2` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccFirebaserulesRelease_BasicRelease|TestAccContainerNodePool_withNetworkConfig </blockquote></details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/bcf2a48c-8118-4bde-be92-1453faafa462/recording/TestAccFirebaserulesRelease_BasicRelease.log)] 

Tests failed during RECORDING mode:
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/bcf2a48c-8118-4bde-be92-1453faafa462/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/bcf2a48c-8118-4bde-be92-1453faafa462/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/bcf2a48c-8118-4bde-be92-1453faafa462/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/bcf2a48c-8118-4bde-be92-1453faafa462/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 109 insertions(+), 10 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 109 insertions(+), 10 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_container_cluster` (254 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_container_cluster" "primary" {
  node_pool {
    network_config {
      network_performance_config {
        total_egress_bandwidth_tier = # value needed
      }
    }
  }
}

```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 109 insertions(+), 10 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 109 insertions(+), 10 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_container_cluster` (254 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_container_cluster" "primary" {
  node_pool {
    network_config {
      network_performance_config {
        total_egress_bandwidth_tier = # value needed
      }
    }
  }
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `3237` 
Passed tests `2904` 
Skipped tests: `331` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccComputeForwardingRule_update|TestAccContainerNodePool_withNetworkConfig </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccComputeForwardingRule_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/80a6aec3-2556-46e7-8aff-d9233f0c65d9/recording/TestAccComputeForwardingRule_update.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/80a6aec3-2556-46e7-8aff-d9233f0c65d9/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/80a6aec3-2556-46e7-8aff-d9233f0c65d9/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/80a6aec3-2556-46e7-8aff-d9233f0c65d9/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/80a6aec3-2556-46e7-8aff-d9233f0c65d9/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 109 insertions(+), 10 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 109 insertions(+), 10 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_container_cluster` (254 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_container_cluster" "primary" {
  node_pool {
    network_config {
      network_performance_config {
        total_egress_bandwidth_tier = # value needed
      }
    }
  }
}

```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 161 insertions(+), 10 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 161 insertions(+), 10 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_container_cluster` (255 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_container_cluster" "primary" {
  node_pool {
    network_config {
      network_performance_config {
        total_egress_bandwidth_tier = # value needed
      }
    }
  }
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `3237` 
Passed tests `2905` 
Skipped tests: `331` 
Affected tests: `1` 

#### Action taken 
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/04af8853-bd24-486e-a597-9afa9e4b49c4/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/04af8853-bd24-486e-a597-9afa9e4b49c4/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/04af8853-bd24-486e-a597-9afa9e4b49c4/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/04af8853-bd24-486e-a597-9afa9e4b49c4/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 107 insertions(+), 8 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 107 insertions(+), 8 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_container_cluster` (254 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_container_cluster" "primary" {
  node_pool {
    network_config {
      network_performance_config {
        total_egress_bandwidth_tier = # value needed
      }
    }
  }
}

```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 76 insertions(+), 10 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 76 insertions(+), 10 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_container_cluster` (254 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_container_cluster" "primary" {
  node_pool {
    network_config {
      network_performance_config {
        total_egress_bandwidth_tier = # value needed
      }
    }
  }
}

```

kawych commented:
@rileykarson Please take another look. Addressed the comments and managed to run the acceptance tests locally (I had to manually edit out the overprovisioning config, seems like that one is allowed only for certain projects...).

modular-magician commented:
#### Tests analytics 
Total tests: `3237` 
Passed tests `2904` 
Skipped tests: `331` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig|TestAccDataSourceGoogleServiceAccountIdToken_impersonation </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDataSourceGoogleServiceAccountIdToken_impersonation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/b8b434ee-42b6-4d93-b84d-613e38817373/recording/TestAccDataSourceGoogleServiceAccountIdToken_impersonation.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/b8b434ee-42b6-4d93-b84d-613e38817373/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/b8b434ee-42b6-4d93-b84d-613e38817373/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/b8b434ee-42b6-4d93-b84d-613e38817373/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/b8b434ee-42b6-4d93-b84d-613e38817373/recording) for each test

rileykarson commented:
Cleared quota, rerunning tests.

/gcbrun

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 4 files changed, 81 insertions(+), 10 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 76 insertions(+), 10 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_container_cluster` (254 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_container_cluster" "primary" {
  node_pool {
    network_config {
      network_performance_config {
        total_egress_bandwidth_tier = # value needed
      }
    }
  }
}

```

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccHealthcareDatasetIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/be331d0a-ed09-4cc8-a2a6-f2fad005490f/recording/TestAccHealthcareDatasetIamPolicy.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/be331d0a-ed09-4cc8-a2a6-f2fad005490f/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/be331d0a-ed09-4cc8-a2a6-f2fad005490f/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/be331d0a-ed09-4cc8-a2a6-f2fad005490f/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/be331d0a-ed09-4cc8-a2a6-f2fad005490f/recording) for each test

rileykarson commented on mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb:
```diff
@@ -3016,19 +3026,35 @@ resource "google_container_node_pool" "with_pco_disabled" {
 	}
   }
   node_config {
+	machine_type = "n2-standard-32"
```
Oh, the issue is from the test alone- we need to use a smaller machine or different type:

```
Insufficient regional quota to satisfy request: resource "N2_CPUS": request requires '96.0' and is short '72.0'. project has a quota of '24.0' with '24.0' available.
```

kawych replied:
Per GKE docs [1] "Tier_1 networking is supported on N2, N2D, C2, C2D, C3, C3D, and M3 VMs that have at least 30 vCPUs.".

If the quota is insufficient for N2s, I doubt there will be sufficient quota for a different type out of those, is it right?  I think we can either skip this test entirely, bump the quota or make it manual-only in some way, what would you recommend?

[1] https://cloud.google.com/compute/docs/networking/configure-vm-with-high-bandwidth-configuration

rileykarson replied:
Could we use a zonal cluster or specify a single node location? That'll reduce us from 96 to 32 CPUs.

kawych replied:
Makes sense, changed to a single node location.

rileykarson commented on mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb:
```diff
@@ -597,6 +599,14 @@ func TestAccContainerNodePool_withNetworkConfig(t *testing.T) {
 			},
+			// edit the updateable network config
+			{
+				Config: testAccContainerNodePool_withNetworkConfig(cluster, np, network, "TIER_UNSPECIFIED"),
```
Have you run this locally? UNSPECIFIED values often get returned as json null (i.e. no field in the response object) and will cause a diff in Terraform. We probably need a different config w/o the `network_performance_config` block at all.

kawych replied:
Yes, it worked for me locally. I thought setting an empty block won't help because TF will not update these values in a case of empty block - this happened for example when I left network_config as an empty block (test failed because the node pool didn't update). Is it a bug?

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 79 insertions(+), 10 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 79 insertions(+), 10 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_container_cluster` (254 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_container_cluster" "primary" {
  node_pool {
    network_config {
      network_performance_config {
        total_egress_bandwidth_tier = # value needed
      }
    }
  }
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `3241` 
Passed tests `2908` 
Skipped tests: `331` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig|TestAccDataprocClusterIamPolicy </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDataprocClusterIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/275c0a64-1f38-445e-a4a8-82ac303f42e0/recording/TestAccDataprocClusterIamPolicy.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/275c0a64-1f38-445e-a4a8-82ac303f42e0/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/275c0a64-1f38-445e-a4a8-82ac303f42e0/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/275c0a64-1f38-445e-a4a8-82ac303f42e0/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/275c0a64-1f38-445e-a4a8-82ac303f42e0/recording) for each test

rileykarson commented:
Submitted QIR

/gcbrun

modular-magician commented:
#### Tests analytics 
Total tests: `3241` 
Passed tests `2908` 
Skipped tests: `331` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig|TestAccDataprocClusterIamPolicy </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDataprocClusterIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/35babd4c-9a0a-4a37-85f9-4b2c27f469c6/recording/TestAccDataprocClusterIamPolicy.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/35babd4c-9a0a-4a37-85f9-4b2c27f469c6/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/35babd4c-9a0a-4a37-85f9-4b2c27f469c6/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/35babd4c-9a0a-4a37-85f9-4b2c27f469c6/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/35babd4c-9a0a-4a37-85f9-4b2c27f469c6/recording) for each test

modular-magician commented:
#### Tests analytics 
Total tests: `3246` 
Passed tests `2913` 
Skipped tests: `331` 
Affected tests: `2` 

#### Action taken 
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig|TestAccGKEHub2Fleet_gkehubFleetBasicExample_update </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccGKEHub2Fleet_gkehubFleetBasicExample_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/cdb850fc-67eb-4d79-802e-da2cde31a255/recording/TestAccGKEHub2Fleet_gkehubFleetBasicExample_update.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{red}{\textsf{Tests failed when rerunning REPLAYING mode:}}$ 
 `TestAccGKEHub2Fleet_gkehubFleetBasicExample_update`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/cdb850fc-67eb-4d79-802e-da2cde31a255/build-log/replaying_build_after_recording/TestAccGKEHub2Fleet_gkehubFleetBasicExample_update_replaying_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/cdb850fc-67eb-4d79-802e-da2cde31a255/replaying_after_recording/TestAccGKEHub2Fleet_gkehubFleetBasicExample_update.log)] 

Tests failed due to non-determinism or randomness when the VCR replayed the response after the HTTP request was made.

Please fix these to complete your PR. If you believe these test failures to be incorrect or unrelated to your change, or if you have any questions, please raise the concern with your reviewer.

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccContainerNodePool_withNetworkConfig`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/cdb850fc-67eb-4d79-802e-da2cde31a255/build-log/recording_build/TestAccContainerNodePool_withNetworkConfig_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/cdb850fc-67eb-4d79-802e-da2cde31a255/recording/TestAccContainerNodePool_withNetworkConfig.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/cdb850fc-67eb-4d79-802e-da2cde31a255/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/cdb850fc-67eb-4d79-802e-da2cde31a255/recording) for each test

rileykarson commented:
Requested more quota- not sure why we were running out here as we should have had enough.

/gcbrun

rileykarson commented:
TestAccGKEHub2Fleet_gkehubFleetBasicExample_update seems unrelated.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 82 insertions(+), 8 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 82 insertions(+), 8 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_container_cluster` (255 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_container_cluster" "primary" {
  node_pool {
    network_config {
      network_performance_config {
        total_egress_bandwidth_tier = # value needed
      }
    }
  }
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `3255` 
Passed tests `2920` 
Skipped tests: `331` 
Affected tests: `4` 

#### Action taken 
<details> <summary>Found 4 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccContainerNodePool_withNetworkConfig|TestAccGKEHub2Fleet_gkehubFleetBasicExample_update|TestAccLoggingProjectSink_updatePreservesCustomWriter|TestAccDataSourceGoogleServiceAccountIdToken_impersonation </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccContainerNodePool_withNetworkConfig`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/dbca17a7-9250-4ae7-aeab-08665ae00090/recording/TestAccContainerNodePool_withNetworkConfig.log)]
`TestAccGKEHub2Fleet_gkehubFleetBasicExample_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/dbca17a7-9250-4ae7-aeab-08665ae00090/recording/TestAccGKEHub2Fleet_gkehubFleetBasicExample_update.log)]
`TestAccDataSourceGoogleServiceAccountIdToken_impersonation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/dbca17a7-9250-4ae7-aeab-08665ae00090/recording/TestAccDataSourceGoogleServiceAccountIdToken_impersonation.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccLoggingProjectSink_updatePreservesCustomWriter`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/dbca17a7-9250-4ae7-aeab-08665ae00090/build-log/recording_build/TestAccLoggingProjectSink_updatePreservesCustomWriter_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/dbca17a7-9250-4ae7-aeab-08665ae00090/recording/TestAccLoggingProjectSink_updatePreservesCustomWriter.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/dbca17a7-9250-4ae7-aeab-08665ae00090/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/dbca17a7-9250-4ae7-aeab-08665ae00090/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 82 insertions(+), 8 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6826-old..auto-pr-6826) ( 3 files changed, 82 insertions(+), 8 deletions(-))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_container_cluster` (271 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_container_cluster" "primary" {
  node_pool {
    network_config {
      network_performance_config {
        total_egress_bandwidth_tier = # value needed
      }
    }
  }
}

```

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$ 
 `TestAccDataprocClusterIamPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/fa434414-2cad-4a0c-85d6-103c621bc9a7/recording/TestAccDataprocClusterIamPolicy.log)]
`TestAccGKEHub2Fleet_gkehubFleetBasicExample_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/fa434414-2cad-4a0c-85d6-103c621bc9a7/recording/TestAccGKEHub2Fleet_gkehubFleetBasicExample_update.log)]
`TestAccDataSourceGoogleServiceAccountIdToken_impersonation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/fa434414-2cad-4a0c-85d6-103c621bc9a7/recording/TestAccDataSourceGoogleServiceAccountIdToken_impersonation.log)] 

##### Rerun these tests in REPLAYING mode to catch issues 

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$ 

---
$\textcolor{green}{\textsf{All tests passed!}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/fa434414-2cad-4a0c-85d6-103c621bc9a7/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6826/artifacts/fa434414-2cad-4a0c-85d6-103c621bc9a7/recording) for each test
